src/box/lua/schema.lua

```diff
@@ -775,6 +783,11 @@ local function update_index_parts(format, parts)
             box.error(box.error.ILLEGAL_PARAMS,
                       "options.parts[" .. i .. "]: type (boolean) is expected")
         end
+        if (not part.is_nullable) and part.exclude_null then
```

Comment:
I think we should made a bit better.
It a user sets `exclude_null=true` and doesn't set `is_nullable` we should internally set `is_nullable=true`.
Only when a user sets `exclude_null=true` and `is_nullable=false` it must be a error.

Reply:
@alyapunov Why it should result in error? I believe it is OK: is_nullable field can't contain nulls, so exclude_null filter just does nothing.

Reply:
I write some thoughts here: now because `exclude_null` turns on `is_nullable`, it implies that `exclude_null` cannot be set in primary key. That means tuple is inserted at least in one index. But if we want to support setting both `exclude_null=true` and `is_nullable=false`, then I suppose primary key should be allowed to have the new option. And then we should think about `on_replace` triggers behavior as Nikita already mentioned.

Reply:
I discussed the problem with Alexander. The thing is if we allowed this configuration (exclude_null=true and is_nullable=false) we would face the 'race' when inserting null to the index. One the one hand, exclude_null throws away tuple containing null (i.e. no error is raised); on the other hand, is_nullable property should raise an error (since we are inserting incompatible value). So depending on which check is fired first, the error may be raised. So to avoid this mess Alexander suggests to make these options incompatible. After this explanation I agree with his point and propose to leave it as is for now.